[MIG] stock_picking_delivery_info_computation: Migration to v12

victor-ditay opened the pull request.

Commit:
[ADD] stock_picking_delivery_info_computation: Improve on calculation weight and volume

pedrobaeza committed:
[IMP] stock_picking_delivery_info_computation: Improve code + missing features

* Better readme
* Improved tests
* Compute volume on sales order confirmation
* Compute volume on backorder creation
* Better code in some places
* Improved views

oca-travis committed:
[UPD] Update stock_picking_delivery_info_computation.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

Commit:
[MIG] stock_picking_delivery_info_computation: Migration to v12

rousseldenis reviewed (approved):
Code review

rafaelbn commented on stock_picking_delivery_info_computation/README.rst:
```diff
@@ -0,0 +1,115 @@
+
+To use this module, you need to:
+
+#. Go to *Inventory > All Transfers* and pick one.
```
All transfers* is not valid.  Only draft, waiting and ready. If the transfer is Done or cancelled the button doesn't appears

pedrobaeza replied:
This is because README must be adapted to v12

rafaelbn reviewed (approved):
Functional test :+1: 

Just review one point in README thanks

victor-ditay commented:
Changes done @rafaelbn

pedrobaeza committed:
[IMP] stock_picking_delivery_info_computation: Allow to edit/compute volume on done pickings

As this field remains isolated, you have both possibilities of recomputing or manually
editing the field in any state.

victor-ditay commented:
@pedrobaeza Commit cherry picked from #213

pedrobaeza commented on stock_picking_delivery_info_computation/views/delivery_view.xml:
```diff
@@ -5,7 +5,7 @@
         <field name="model">stock.picking</field>
         <field name="inherit_id" ref="delivery.view_picking_withcarrier_out_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='number_of_packages']" position="after">
```
We should add this field as it has been removed and it's important for the aim on the module: delivery info.

pedrobaeza reviewed (changes requested):
One important thing for keeping information consistent

victor-ditay commented:
Changes done @pedrobaeza

pedrobaeza commented:
/ocabot merge
